pkg/runner/gitlab-api-runner/runner.go

```diff
@@ -104,7 +104,29 @@ func (r *RESTRunner) readInputData(in runner.StartInput) (Input, error) {
 	}, nil
 }
 
-func (r *RESTRunner) renderOutput(artifactTemplate string, data map[string]interface{}) ([]byte, error) {
+func (r *RESTRunner) gitlabClientForAuth(auth Auth, opts []gitlab.ClientOptionFunc) (*gitlab.Client, error) {
+	// access token
+	token := auth.Token
+	if token != nil {
+		return gitlab.NewClient(
+			*token,
+			opts...,
+		)
+	}
+
+	// basic credentials
+	if auth.Basic != nil {
+		return gitlab.NewBasicAuthClient(
+			auth.Basic.Username,
+			auth.Basic.Password,
+			opts...,
+		)
+	}
+
+	return nil, errors.New("no token or basic credentials provided")
```

Comment:
IMO it will be good to return an error when both basic and token are provided, this may reduce a lot of time spent on debugging in case of some misconfiguration :D